linker/linker.cpp

```diff
@@ -366,7 +371,8 @@ static bool realpath_fd(int fd, std::string* realpath) {
 // Intended to be called by libc's __gnu_Unwind_Find_exidx().
 _Unwind_Ptr do_dl_unwind_find_exidx(_Unwind_Ptr pc, int* pcount) {
   for (soinfo* si = solist_get_head(); si != 0; si = si->next) {
-    if ((pc >= si->base) && (pc < (si->base + si->size))) {
+    if (((pc >= si->base) && (pc < (si->base + si->size))) ||
+        si->find_rand_segment_info(pc) != nullptr) {
```

Comment:
I don't really mind making one, but I don't think it would add much. It's pretty clear (to me at least?) that returning null means it couldn't find a rand segment. I will definitely rename that function to make it clear that takes a loaded memory address rather than a file address.